src/vm/contexts.rs

```diff
@@ -452,8 +452,10 @@ impl <'a> OwnedEnvironment <'a> {
     pub fn stx_faucet(&mut self, recipient: &PrincipalData, amount: u128) {
         self.execute_in_env(recipient.clone().into(),
                             |env| {
-                                let bal = env.global_context.database.get_account_stx_balance(recipient);
-                                env.global_context.database.set_account_stx_balance(recipient, bal + amount);
+                                let mut balance = env.global_context.database.get_account_stx_balance(recipient);
+                                let block_height = env.global_context.database.get_current_burnchain_block_height();
+                                balance.credit(amount, block_height as u64).unwrap();
```

Comment:
Should this be `expect()`?

Reply:
This function is only available in `tests`, but sure. Addressed with https://github.com/blockstack/stacks-blockchain/pull/1850/commits/0c76e29ed9431ef3950a5ccf558f0dc568a617fc